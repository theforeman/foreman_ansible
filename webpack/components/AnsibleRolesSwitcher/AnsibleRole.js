import React from 'react';
import { ListView, Icon } from 'patternfly-react';
import classNames from 'classnames';

import './styles.scss';

const clickHandler = (onClick, role) => (event) => {
  event.preventDefault();
  onClick(role);
};


const actionButton = (icon, onClick, role) => (
  <button
    href='#'
    onClick={clickHandler(onClick, role)}
    className="role-add-remove-btn"
  >
    <Icon className='fa-2x' type='fa' name={icon} />
  </button>
);

const AnsibleRole = ({ role, icon, onClick }) => (
  <ListView.Item
    id={role.id}
    className={classNames('listViewItem--listItemVariants', { 'ansible-role-disabled': role.inherited })}
    heading={role.name}
    actions={role.inherited ? '' : actionButton(icon, onClick, role)}
    stacked
  />
);

export default AnsibleRole;
