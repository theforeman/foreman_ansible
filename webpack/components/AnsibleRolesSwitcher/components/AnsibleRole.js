import React from 'react';
import { ListView, Tooltip, OverlayTrigger } from 'patternfly-react';
import classNames from 'classnames';
import { translate as __ } from 'foremanReact/common/I18n';

import AnsibleRoleActionButton from './AnsibleRoleActionButton';
import '../AnsibleRolesSwitcher.scss';

const AnsibleRole = ({ role, icon, onClick, resourceName }) => {
  const text =
    resourceName === 'hostgroup'
      ? __('This Ansible role is inherited from parent host group')
      : __('This Ansible role is inherited from host group');

  const tooltip = (
    <Tooltip id={role.id}>
      <span>{text}</span>
    </Tooltip>
  );

  const clickHandler = (onClickFn, ansibleRole) => event => {
    event.preventDefault();
    onClickFn(ansibleRole);
  };

  const listItem = (click = undefined) => (
    <ListView.Item
      id={role.id}
      className={classNames('listViewItem--listItemVariants', {
        'ansible-role-disabled': role.inherited,
        'ansible-role-movable': !role.inherited,
      })}
      heading={role.name}
      actions={
        role.inherited ? (
          ''
        ) : (
          <AnsibleRoleActionButton icon={icon} role={role} />
        )
      }
      stacked
      onClick={typeof click === 'function' ? click(onClick, role) : click}
    />
  );

  if (role.inherited) {
    return (
      <OverlayTrigger overlay={tooltip} placement="top">
        {listItem()}
      </OverlayTrigger>
    );
  }

  return listItem(clickHandler);
};

export default AnsibleRole;
