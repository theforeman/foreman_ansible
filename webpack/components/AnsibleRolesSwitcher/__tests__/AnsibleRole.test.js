import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AnsibleRole from '../AnsibleRole';

describe('AnsibleRole', () => {
  const noop = () => {};
  it('should render a role to add', () => {
    const page = shallow(<AnsibleRole
        role={{ name: 'test.role', id: 5 }}
        icon='fa fa-plus-circle'
        onClick={noop}
      />);

    expect(toJson(page)).toMatchSnapshot();
  });

  it('should render a role to remove', () => {
    const page = shallow(<AnsibleRole
        role={{ name: 'test.role', id: 5 }}
        icon='fa fa-minus-circle'
        onClick={noop}
      />);

    expect(toJson(page)).toMatchSnapshot();
  });

  it('should render inherited role to remove', () => {
    const page = shallow(<AnsibleRole
        role={{ name: 'test.role', id: 5, inherited: true }}
        icon='fa fa-minus-circle'
        onClick={noop}
      />);

    expect(toJson(page)).toMatchSnapshot();
  });
});
