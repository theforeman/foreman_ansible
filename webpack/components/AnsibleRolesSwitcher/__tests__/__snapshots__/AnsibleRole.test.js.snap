// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleRole should render a role to add 1`] = `
<ListViewItem
  actions={
    <button
      className="role-add-remove-btn"
      href="#"
      onClick={[Function]}
    >
      <Icon
        className="fa-2x"
        name="fa fa-plus-circle"
        type="fa"
      />
    </button>
  }
  additionalInfo={null}
  checkboxInput={null}
  className="listViewItem--listItemVariants"
  compoundExpand={false}
  compoundExpanded={false}
  description={null}
  heading="test.role"
  hideCloseIcon={false}
  id={5}
  initExpanded={false}
  leftContent={null}
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
/>
`;

exports[`AnsibleRole should render a role to remove 1`] = `
<ListViewItem
  actions={
    <button
      className="role-add-remove-btn"
      href="#"
      onClick={[Function]}
    >
      <Icon
        className="fa-2x"
        name="fa fa-minus-circle"
        type="fa"
      />
    </button>
  }
  additionalInfo={null}
  checkboxInput={null}
  className="listViewItem--listItemVariants"
  compoundExpand={false}
  compoundExpanded={false}
  description={null}
  heading="test.role"
  hideCloseIcon={false}
  id={5}
  initExpanded={false}
  leftContent={null}
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
/>
`;

exports[`AnsibleRole should render inherited role to remove 1`] = `
<ListViewItem
  actions=""
  additionalInfo={null}
  checkboxInput={null}
  className="listViewItem--listItemVariants ansible-role-disabled"
  compoundExpand={false}
  compoundExpanded={false}
  description={null}
  heading="test.role"
  hideCloseIcon={false}
  id={5}
  initExpanded={false}
  leftContent={null}
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
/>
`;
